canopeum_backend/canopeum_backend/models.py

```diff
@@ -1,22 +1,28 @@
 from datetime import datetime, timedelta
-from typing import ClassVar
+from typing import TYPE_CHECKING, ClassVar
 
 import pytz
 from django.contrib.auth.models import AbstractUser
 from django.db import models
+from rest_framework.request import Request as drf_Request
+
+# Pyright won't be able to infer all types here, see:
+# https://github.com/typeddjango/django-stubs/issues/579
+# https://github.com/typeddjango/django-stubs/issues/1264
+# For now we have to rely on the mypy plugin
 
 
 class RoleName(models.TextChoices):
     USER = "User"
     SITEMANAGER = "SiteManager"
     MEGAADMIN = "MegaAdmin"
 
-    def from_string(self, value):
-        if value == self.MEGAADMIN:
-            return self.MEGAADMIN
-        if value == self.SITEMANAGER:
-            return self.SITEMANAGER
-        return self.USER
+    @classmethod
+    def from_string(cls, value: str):
```

Comment:
CC @NicolasDontigny This is how you do a classmethod in Python (ie: a method you call on the class itself, this is technically different than a static method, but this would be a static method in other languages)